### PR TITLE
RUN-5382  Iframe names are incorrect with the v2 api

### PR DESCRIPTION
**Pull request by aziz512**
As per description of the issue:

"When calling getCurrentSync().getInfo the identity object returned should not share the name of the host window. This appears only to be broken in the v2  api. We need to make sure that the name returned matches the actual name (a guid produced by the runtime) that is used for routing."

This PR ensures that correct frame identity is used when initializing and working with iframes.
@rdepena  @datamadic @pbaize

### CODE REVIEW

File: src/api/frame/frame.ts [hunk 1]
```diff
@@ -18,6 +19,12 @@ export interface FrameInfo {
  */
 // tslint:disable-next-line
 export default class _FrameModule extends Base {
+    public frameIdentity: any;
```
Comment (dhamberlin) [inline src/api/frame/frame.ts]: Could we use a more specific type here? I think this will be an `Identity`?

File: src/api/frame/frame.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+declare var fin: any;
```
Comment (dhamberlin) [inline src/api/frame/frame.ts]: Hipster nitpick: `var` is so 2014... cool kids use `const`

Not actually a blocker obviously.

File: src/api/frame/frame.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
...
 import { Base, EmitterBase } from '../base';
```
Comment (licui3936) [inline src/api/frame/frame.ts]: can you move fin related item to the openfin render api file under environment? so no need to define here.

File: src/api/frame/frame.ts [hunk 4]
```diff
@@ -18,6 +19,12 @@ export interface FrameInfo {
  */
 // tslint:disable-next-line
 export default class _FrameModule extends Base {
+    public frameIdentity: any;
+    constructor(wire: Transport) {
+        super(wire);
+        this.frameIdentity = fin.__internal_.entityInfo;
```
Comment (licui3936) [inline src/api/frame/frame.ts]: Can you move this part to environment as i mentioned above.  So it will not break when we run it in node environment.

Comment (aziz512) [reply src/api/frame/frame.ts]: Fixed :+1:

File: src/environment/openfin-env.ts [hunk 5]
```diff
@@ -72,4 +72,8 @@ export default class OpenFinEnvironment implements Environment {
     public getWebWindow = (identity: Identity): Window => {
         return fin.__internal_.getWebWindow(identity.name);
     }
+
+    public getCurrentEntityIdentity = (): Identity => {
+        return fin.__internal_.entityInfo;
```
Comment (datamadic) [inline src/environment/openfin-env.ts]: @pbaize is this still OK in browser view world?

**Pull request merged**